commit 302896f06cbbff1453deba0a71e20f1b38919f42
Author: P33M <p33m@github.com>
Date:   Tue Jan 9 15:16:35 2018 +0000

    dwc_otg: don't unconditionally force host mode in dwc_otg_cil_init()
    
    Add the ability to disable force_host_mode for those that want to use
    dwc_otg in both device and host modes.

diff --git a/drivers/usb/host/dwc_otg/dwc_otg_cil.c b/drivers/usb/host/dwc_otg/dwc_otg_cil.c
index 38abd0b1f034..14300ad0f2af 100644
--- a/drivers/usb/host/dwc_otg/dwc_otg_cil.c
+++ b/drivers/usb/host/dwc_otg/dwc_otg_cil.c
@@ -61,6 +61,8 @@
 #include "dwc_otg_regs.h"
 #include "dwc_otg_cil.h"
 
+extern bool cil_force_host;
+
 static int dwc_otg_setup_params(dwc_otg_core_if_t * core_if);
 
 /**
@@ -192,7 +194,10 @@ dwc_otg_core_if_t *dwc_otg_cil_init(const uint32_t * reg_base_addr)
 		core_if->hptxfsiz.d32 =
 		DWC_READ_REG32(&core_if->core_global_regs->hptxfsiz);
 		gusbcfg.d32 =  DWC_READ_REG32(&core_if->core_global_regs->gusbcfg);
-		gusbcfg.b.force_host_mode = 1;
+		if (cil_force_host)
+			gusbcfg.b.force_host_mode = 1;
+		else
+			gusbcfg.b.force_host_mode = 0;
 		DWC_WRITE_REG32(&core_if->core_global_regs->gusbcfg, gusbcfg.d32);
 		dwc_mdelay(100);
 	}
diff --git a/drivers/usb/host/dwc_otg/dwc_otg_driver.c b/drivers/usb/host/dwc_otg/dwc_otg_driver.c
index af123d01f3ef..e945900c503c 100644
--- a/drivers/usb/host/dwc_otg/dwc_otg_driver.c
+++ b/drivers/usb/host/dwc_otg/dwc_otg_driver.c
@@ -247,6 +247,9 @@ bool fiq_fsm_enable = true;
 //Bulk split-transaction NAK holdoff in microframes
 uint16_t nak_holdoff = 8;
 
+//Force host mode during CIL re-init
+bool cil_force_host = true;
+
 unsigned short fiq_fsm_mask = 0x0F;
 
 unsigned short int_ep_interval_min = 0;
@@ -1403,6 +1406,10 @@ MODULE_PARM_DESC(int_ep_interval_min, "Clamp high-speed Interrupt endpoints to a
 					"0..1 = Use endpoint default\n"
 					"2..n = Minimum interval n microframes. Use powers of 2.\n");
 
+module_param(cil_force_host, bool, 0644);
+MODULE_PARM_DESC(cil_force_host, "On a connector-ID status change, "
+					"force Host Mode regardless of OTG state.");
+
 /** @page "Module Parameters"
  *
  * The following parameters may be specified when starting the module.
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1164_302896f06cbbff1453deba0a71e20f1b38919f42.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1164_302896f06cbbff1453deba0a71e20f1b38919f42.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
